sql/core/src/test/scala/org/apache/spark/sql/execution/UnsafeExternalSortSuite.scala

```diff
@@ -97,7 +93,8 @@ class UnsafeExternalSortSuite extends SparkPlanTest with BeforeAndAfterAll {
         inputDf,
         UnsafeExternalSort(sortOrder, global = true, _: SparkPlan, testSpillFrequency = 23),
         Sort(sortOrder, global = true, _: SparkPlan),
-        sortAnswers = false
+        sortAnswers = false,
+        compareStrings = true
```

Comment:
Instead of comparing as String, could you update the Row.equals() to handle NaN (we need to do this eventually)?